diff --git a/src/dawn/native/CMakeLists.txt b/src/dawn/native/CMakeLists.txt
index 70dbfb2f51..ba34075395 100644
--- a/src/dawn/native/CMakeLists.txt
+++ b/src/dawn/native/CMakeLists.txt
@@ -526,10 +526,11 @@ if (DAWN_ENABLE_METAL)
         "metal/UtilsMetal.mm"
         "Surface_metal.mm"
     )
-    if (DAWN_TARGET_MACOS OR IOS)
+    if (DAWN_TARGET_MACOS)
         list(APPEND conditional_private_platform_depends "-framework Cocoa")
     endif()
     list(APPEND conditional_private_platform_depends
+        "-framework Foundation"
         "-framework IOKit"
         "-framework IOSurface"
         "-framework QuartzCore"
