diff --git a/src/cmake/BundleLibraries.cmake b/src/cmake/BundleLibraries.cmake
index 046a6af10d..5a63ac3d6d 100644
--- a/src/cmake/BundleLibraries.cmake
+++ b/src/cmake/BundleLibraries.cmake
@@ -66,8 +66,11 @@ function(bundle_libraries output_target)
     endif()
   endforeach()
 
-  add_library(${output_target} SHARED ${all_objects})
-
+  if(${CMAKE_SYSTEM_NAME} STREQUAL "iOS" OR ${CMAKE_SYSTEM_NAME} STREQUAL "visionOS" OR ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+    add_library(${output_target} STATIC ${all_objects})
+  else()
+    add_library(${output_target} SHARED ${all_objects}) # Default to STATIC for other platforms
+  endif()
   add_dependencies(${output_target} ${ARGN})
 
 endfunction()
